Islandora/islandora_openseadragon
# Updated OpenSeadragon to the latest version to support IIIF 2.1

**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-961)
- Other Relevant Links (Google Groups discussion, related pull requests, Release pull requests, etc.)

https://groups.google.com/forum/#!msg/islandora/aTLDDhCm50A/nT40KRBYvAEJ
# What does this Pull Request do?

This pull updates OpenSeadragon module to support the latest version of OpenSeadraon and with that the latest version of [IIIF](http://iiif.io/api/image/2.0/compliance/#status-of-this-document). It makes a number of changes to the exposed properties of OpenSeadragon for the administrators of the site such that they can customised most aspects of it's display. Also additional documentation is provided on how to setup Loris as the IIIF based Images server for use by OpenSeadragon. While Loris isn't at this time the most ideal image server for our use, this work helps build a foundation for us to move towards full IIIF support across the supported viewers. 
# What's new?
- Changes to the structure of `islandora_openseadragon_settings` variable.
- Theme function callback parameters of `islandora_openseadragon_viewer`.
- Drush install plugin now uses latest version of OpenSeadragon.
# How should this be tested?

If you have an _existing_ install it's required to update Openseadragon to it's latest version. You can do this quickly with the provided Drush command.

``` bash
drush openseadragonplugin
```

Also run Drush update to clear out the old variable usage.

A description of what steps someone could take to:
- Verify changes work with existing uses of the Islandora OpenSeadragon Module
- Test Loris support against the documentation.
# Additional Notes:

It's important that OpenSeadragon be updated to it's latest version (2.2.1) as the previous required version '0.9' is not longer supported. This might have an affect on some users.

This change could have impact on existing code for users in which called internal functions that have moved or relied on the existing interfaces of theme function in their current implementation. There doesn't appear to be such use in the core modules though.
# Interested parties

@Islandora/7-x-1-x-committers
